### PR TITLE
Course implementation

### PR DESCRIPTION
**Pull request by zerubeus**
# بسم الله الرحمن الرحيم

## What's new in this PR?

fixes

لله الحمد والمنة

### CODE REVIEW

File: src/components/modules/AppChapter/MainChapter.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+  const navigationButtons = (
+    <NavigationButtons isChapterComplete next={nextChapter} toggleCompleteChapter={toggleCompleteChapter} />
+  )
+
+  const drawer = (
+    <div>
+      <Divider />
+      <List>
+        {['Inbox', 'Starred', 'Send email', 'Drafts'].map((text, index) => (
```
Comment (abumalick) [inline src/components/modules/AppChapter/MainChapter.tsx]: lol

File: src/components/modules/Chapter/index.tsx [hunk 2]
```diff
@@ -32,26 +36,29 @@ const Chapter: React.FC<IChapterProps> = ({ data, pageContext }) => {
...
 
   if (!chapterStrings) {
     // TODO: Log to Sentry
     return <div>{"Error, we didn't find any strings for this chapter ..."}</div>
   }
-  const otherLocalePath = `${data.otherLocaleTranslations.localePath}${data.api.chapter.course.track.slug}/${data.api.chapter.course.slug}/${data.api.chapter.slug}`
+  const otherLocalePath = `${data?.otherLocaleTranslations.localePath}${data?.api.chapter.course.track.slug}/${data?.api.chapter.course.slug}/${data?.api.chapter.slug}`
 
-  const navigationButtons = (
+  const navigationButtons = t && pageContext && (
```
Comment (abumalick) [inline src/components/modules/Chapter/index.tsx:50]: this condition is a bit strange

File: src/pages/app.tsx [hunk 3]
```diff
@@ -61,18 +71,22 @@ const App = () => {
...
-          // maxWidth="sm"
-          >
-            <Router basepath="/app">
-              <TracksContainer path="/" />
-              <TrackContainer path=":trackId" />
-              <Profile path="/profile" me={user.authUser} signOut={user.signOut} />
+          <Container sx={{ mt: '64px' }} maxWidth="xl">
+            <Router basepath="app">
+              <GenericRouteWrapper path="tracks">
+                <Fragment>
```
Comment (abumalick) [inline src/pages/app.tsx:77]: What do you think about puting the placement inside GenericRouteWrapper?

**Review by abumalick — APPROVED** [review]
LGTM, good work 👍

File: src/components/modules/AppChapter/MainChapter.tsx (hunk 1 above)
Comment (zerubeus) [reply src/components/modules/AppChapter/MainChapter.tsx]: XD

File: src/components/modules/Chapter/index.tsx (hunk 2 above)
Comment (zerubeus) [reply src/components/modules/Chapter/index.tsx:50]: This was to fix ts linting, it was broke in this file

File: src/pages/app.tsx (hunk 3 above)
Comment (zerubeus) [reply src/pages/app.tsx:77]: What is the placement?

Comment (abumalick) [reply src/pages/app.tsx:77]: `Fragment`, sorry

**Pull request merged**